cmd/tendermint/commands/reset.go

```diff
@@ -13,81 +13,71 @@ import (
 	"github.com/tendermint/tendermint/types"
 )
 
-// MakeResetAllCommand constructs a command that removes the database of
+// MakeResetCommand constructs a command that removes the database of
 // the specified Tendermint core instance.
-func MakeResetAllCommand(conf *config.Config, logger log.Logger) *cobra.Command {
-	var keyType string
+func MakeResetCommand(conf *config.Config, logger log.Logger) *cobra.Command {
+	var (
+		resetAllFlag       bool
+		resetPrivValFlag   bool
+		resetPeerStoreFlag bool
+		keyType            string
+	)
 
 	cmd := &cobra.Command{
-		Use:   "unsafe-reset-all",
-		Short: "(unsafe) Remove all the data and WAL, reset this node's validator to genesis state",
+		Use:   "reset",
+		Short: "Remove all tendermint data, resetting to genesis state.",
+		Long: `Removes block, state and evidence store. Does not alter priv-validator state or your peer store.
+If resetting, don't forget to reset application state as well.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return resetAll(conf.DBDir(), conf.PrivValidator.KeyFile(),
-				conf.PrivValidator.StateFile(), logger, keyType)
+			if resetAllFlag {
+				return ResetAll(conf.DBDir(), conf.PrivValidator.KeyFile(),
+					conf.PrivValidator.StateFile(), logger, keyType)
+			}
+			if resetPrivValFlag {
+				return ResetFilePV(conf.PrivValidator.KeyFile(), conf.PrivValidator.StateFile(), logger, keyType)
+			}
+			if resetPeerStoreFlag {
+				return ResetPeerStore(conf.DBDir())
+			}
+			return ResetState(conf.DBDir(), logger)
 		},
 	}
 	cmd.Flags().StringVar(&keyType, "key", types.ABCIPubKeyTypeEd25519,
 		"Key type to generate privval file with. Options: ed25519, secp256k1")
+	cmd.Flags().BoolVar(&resetAllFlag, "all", false, "Resets all block data and priv validator state. Only use in testing")
```

Comment:
Can we prefix with `unsafe-` to indicate that it really should not be done in production? (Yes, I know people may still do it, but at least to mark that maybe it should not be done in production).

Possibly also consider adding a brief interactive prompt to communicate to the user that what they are doing is destructive and cannot be reversed of the form:

`You are about to completely remove all block data and validator key state. This cannot be reversed. Are you sure you wish to continue [y/N]?` Where 'n' is the default.

Reply:
Based on experience the `unsafe-` prefix hasn't really been a successful deterrent. I would be more in favour of having a prompt that the user needs to confirm

Reply:
The main potential advantage of the unsafe prefix is not as a deterrent but as a clear documentation for the reader of a script (say) that the author of that script chose to take a documented dangerous action. Another useful trick could be to make the name annoyingly long (e.g., `reset-data-yes-i-know-this-is-dangerous`). That said, I'm not too exercised about the name.

Reply:
lol I love this name `reset-data-yes-i-know-this-is-dangerous`

Reply:
And keep the capitalization. That's a brilliant touch. 😂

Reply:
I think you all have this well in hand, but I agree with having some kind of `unsafe` or similar prefix as much to be able to say "well, we did say it was unsafe" when people get slashed for double signing.